### PR TITLE
add median of two sorted array

### PR DESCRIPTION
**Pull request by BamaCharanChhandogi**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Comment by BamaCharanChhandogi** [comment]
This is the first PR of the Median of two sorted arrays. 
@cclauss please check.

File: data_structures/arrays/median_two_array.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        if total % 2 == 1:
+            # If the total number of elements is odd, return the
+            # middle element as the median.
+            return float(merged[total // 2])
+        else:
+            # If the total number of elements is even, calculate
+            # the average of the two middle elements as the median.
+            middle1 = merged[total // 2 - 1]
+            middle2 = merged[total // 2]
+            return (float(middle1) + float(middle2)) / 2.0
```
Comment (cclauss) [inline data_structures/arrays/median_two_array.py:42]:
```suggestion
        # Merge the arrays into a single sorted array.
        merged = sorted(nums1 + nums2)
        length = len(merged)
        if total % 2 == 1:  # If the total number of elements is odd
            return float(merged[total // 2])  # then return the middle element

        # If the total number of elements is even, calculate
        # the average of the two middle elements as the median.
        middle1 = merged[total // 2 - 1]
        middle2 = merged[total // 2]
        return (float(middle1) + float(middle2)) / 2.0
```

File: data_structures/arrays/median_two_array.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        Example:
+        >>> sol = Solution()
+        >>> sol.find_median_sorted_arrays([1, 3], [2])
+        2.0
+
+        >>> sol.find_median_sorted_arrays([1, 2], [3, 4])
+        2.5
+
+        >>> sol.find_median_sorted_arrays([0, 0], [0, 0])
+        0.0
```
Comment (cclauss) [inline data_structures/arrays/median_two_array.py:22]: Please add tests:
```
>>> find_median_sorted_arrays([], [])
>>> find_median_sorted_arrays([], [1])
>>> find_median_sorted_arrays([-1_000], (1_000))
>>> find_median_sorted_arrays((-1.1, -2.2), [-3.3, -4.4])

**Review by cclauss — CHANGES_REQUESTED** [review]
Under what circumstances would this algorithm be used?

File: data_structures/arrays/median_two_array.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
+class Solution:
+    def find_median_sorted_arrays(self, nums1: list[int], nums2: list[int]) -> float:
```
[the remark this answers is not in the document]
Comment (BamaCharanChhandogi) [reply data_structures/arrays/median_two_array.py]: done

File: data_structures/arrays/median_two_array.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        """
+        Find the median of two sorted arrays.
+
+        Args:
+            nums1: The first sorted array.
+            nums2: The second sorted array.
+
+        Returns:
+            The median of the two sorted arrays.
```
[the remark this answers is not in the document]
Comment (BamaCharanChhandogi) [reply data_structures/arrays/median_two_array.py]: done

File: data_structures/arrays/median_two_array.py (hunk 1 above)
Comment (BamaCharanChhandogi) [reply data_structures/arrays/median_two_array.py]: done

File: data_structures/arrays/median_two_array.py (hunk 2 above)
Comment (BamaCharanChhandogi) [reply data_structures/arrays/median_two_array.py:20]: done

**Comment by BamaCharanChhandogi** [comment]
Thanks for the guidance.

**Comment by cclauss** [comment]
https://github.com/TheAlgorithms/Python/pull/9386#pullrequestreview-1652185734

**Comment by BamaCharanChhandogi** [comment]
I think I have resolved all.

**Comment by cclauss** [comment]
https://github.com/TheAlgorithms/Python/pull/9386#pullrequestreview-1652185734

**Comment by BamaCharanChhandogi** [comment]
You are saying I will add  the answer to Your question in docstring.
Right?

**Comment by BamaCharanChhandogi** [comment]
@cclauss please check

**Comment by BamaCharanChhandogi** [comment]
Hello @cclauss, Please tell me if anything needed.

**Comment by BamaCharanChhandogi** [comment]
@cclauss please check

**Comment by BamaCharanChhandogi** [comment]
@cclauss check please

**Review by cclauss — APPROVED** [review]

**Pull request merged**